packages/hop-node/src/watchers/xDaiBridgeWatcher.ts

```diff
@@ -101,89 +56,110 @@ class xDaiBridgeWatcher extends BaseWatcher {
     }
   }
 
-  async start () {
-    this.started = true
+  async handleCommitTxHash (commitTxHash: string, transferRootHash: string, logger: Logger) {
+    logger.debug(
+      `attempting to send relay message on xdai for commit tx hash ${commitTxHash}`
+    )
+    await this.handleStateSwitch()
+    if (this.isDryOrPauseMode) {
+      logger.warn(`dry: ${this.dryMode}, pause: ${this.pauseMode}. skipping relayXDomainMessage`)
+      return
+    }
+    await this.db.transferRoots.update(transferRootHash, {
+      sentConfirmTxAt: Date.now()
+    })
+
     try {
-      const l1Amb = getL1Amb(this.tokenSymbol)
-      const l2Amb = getL2Amb(this.tokenSymbol)
-
-      this.logger.debug(`xDai ${this.tokenSymbol} bridge watcher started`)
-      while (true) {
-        if (!this.started) {
-          return
-        }
-        const blockNumber = await l2Amb.provider.getBlockNumber()
-        const events = await l2Amb.queryFilter(
-          l2Amb.filters.UserRequestForSignature(),
-          blockNumber - 100
-        )
-
-        for (const event of events) {
-          try {
-            const result = await executeExitTx(event, this.tokenSymbol)
-            if (result == null) {
-              continue
-            }
-            const { tx, msgHash } = result
-            this.logger.debug(`executeSignatures. tx hash: ${tx.hash} messageHash: ${msgHash}`)
-          } catch (err) {
-            this.logger.error('tx error:', err.message)
-          }
-        }
-        await wait(10 * 1000)
+      const result = await this.relayXDomainMessage(commitTxHash)
+      if (result == null) {
+        logger.error('no result returned from exit tx')
+        return
       }
+      const { tx } = result
+      const msg = `sent chainId ${this.bridge.chainId} confirmTransferRoot L1 exit tx ${tx.hash}`
+      logger.info(msg)
+      this.notifier.info(msg)
     } catch (err) {
-      this.logger.error('xDai bridge watcher error:', err)
-      this.quit()
+      logger.error(err)
+      throw err
```

Comment:
if the error is being propagated up and then logged then it's going to get logged twice

- suggestion: remove try/catch and logger.error, or add additional context to the logger.error here

Reply:
Good call. We are using a `try...catch` for Optimism to explicitly handle certain errors. Since we are not explicitly handling errors in other watchers, I removed the `try...catch`es from them in eea40482dd494755352531db496125963da10f27